exchange/utils.go

```diff
@@ -302,14 +313,45 @@ func buildRequestExtForBidder(bidder string, requestExt json.RawMessage, request
 
 	// Update Ext With Prebid Json
 	extMap := make(map[string]json.RawMessage)
-	if err := json.Unmarshal(requestExt, &extMap); err != nil {
-		return nil, err
+	if len(requestExt) != 0 {
+		if err := json.Unmarshal(requestExt, &extMap); err != nil {
+			return nil, err
+		}
 	}
 	extMap["prebid"] = prebidJson
 
 	return json.Marshal(extMap)
 }
 
+func buildRequestExtAlternateBidderCodes(bidder string, accABC config.AlternateBidderCodes, reqABC *openrtb_ext.ExtAlternateBidderCodes) *openrtb_ext.ExtAlternateBidderCodes {
```

Comment:
After looking more closely at the requirements, I believe we should aim for the following behavior:
- If `alternatebiddercodes` is not defined in the account or on the request, the field should be omitted from all bidder requests.
- If `alternatebiddercodes` is defined either in the account or on the request but its `enabled` is not set, the `alternatebiddercodes` field should be added to the bidder request with the default value of false.
- The `alternatebiddercodes` object set on a bidder request should only contain its own bidder object in the `bidders` field and it should match the bidder object in the account/request it was sourced from. Note an undefined map or slice in the original request or account will appear as `null` in a bidder request (we will need to get rid of `omitempty`), so that we can distinguish between an empty array/map and an undefined array/map.

Given the above, I'd like to suggest a couple of things here:
1) In `config/accounts.go`, can we make the field `AlternateBidderCodes` in the `Account` struct of type `*AlternateBidderCodes`?
2) Let's get rid of the `if` block on line 348 that sets `alternateBidderCodes = nil`.
3) I think this function ultimately looks like this:
```
        if reqABC != nil {
		alternateBidderCodes := &openrtb_ext.ExtAlternateBidderCodes{
			Enabled: reqABC.Enabled,
		}
		if bidderCodes, ok := reqABC.Bidders[bidder]; ok {
			alternateBidderCodes.Bidders = map[string]openrtb_ext.ExtAdapterAlternateBidderCodes{
				bidder: bidderCodes,
			}
		}
		return alternateBidderCodes
	}

	if accABC != nil {
		alternateBidderCodes := &openrtb_ext.ExtAlternateBidderCodes{
			Enabled: accABC.Enabled,
		}
		if bidderCodes, ok := accABC.Bidders[bidder]; ok {
			alternateBidderCodes.Bidders = map[string]openrtb_ext.ExtAdapterAlternateBidderCodes{
				bidder: bidderCodes,
			}
		}
		return alternateBidderCodes
	}
	return nil
```

Reply:
done. made `accABC` pointer to detect its presence.